Src/Compilers/Core/MSBuildTask/BuildClient.cs

```diff
@@ -48,6 +48,81 @@ private static string GetExpectedServerExeDir()
         }
 
         /// <summary>
+        /// Run a compilation through the compiler server and print the output
+        /// to the console. If the compiler server fails, run the fallback
+        /// compiler.
+        /// </summary>
+        public static int RunWithConsoleOutput(
+            string[] args,
+            RequestLanguage language,
+            Func<string[], int> fallbackCompiler)
+        {
+            var errorMessage = CommandLineParser.CheckClientArgsForErrors(args);
+
+            if (errorMessage != null)
+            {
+                Console.Out.WriteLine(errorMessage);
+                return 1;
+            }
+
+            var responseTask = TryRunServerCompilation(
+                language,
+                Environment.CurrentDirectory,
+                args,
+                default(CancellationToken),
+                libEnvVariable: Environment.GetEnvironmentVariable("LIB"));
+
+            responseTask.Wait();
+
+            int exitCode;
+            var response = responseTask.Result;
```

Comment:
It's probably possible. Do you think it's worth trying to put in an exception filter fail fast somewhere here? I don't see a great place to put a `try` block (maybe in TryRunServerCompilation?) -- do you have any suggestions?